kernel/sched.c

```diff
@@ -1594,6 +1594,12 @@ static int cpu_mask_mod(k_tid_t thread, uint32_t enable_mask, uint32_t disable_m
 {
 	int ret = 0;
 
+#if defined(CONFIG_ASSERT) && defined(CONFIG_SCHED_CPU_MASK_PIN_ONLY)
```

Comment:
```suggestion
#if defined(CONFIG_SCHED_CPU_MASK_PIN_ONLY)
```
https://docs.zephyrproject.org/latest/kernel/services/other/fatal.html

Reply:
I thought about it, but that is the pattern used few line bellow

Reply:
Should fix the other usage then.  The __ASSERT() macro is always defined (though potentially to an empty string), you don't need to test for anything to use it.

Reply:
The `CONFIG_ASSERT` below is there for a reason: `int m = ...`. No such reason here.

There are dozens of other `__ASSERT`s in this file and they almost never use `CONFIG_ASSERT`.